### ISSUE TITLE
Unallocated CPU resources are not being calculated correctly

### ISSUE THREAD

**Issue by jorgemarey** [hashicorp/nomad#23371]
### Nomad version
Nomad v1.8.0

### Issue

When declaring reserved CPU compute unallocated resources are not being correctly calculared.

### Reproduction steps

1)
❯ cat config.hcl 
```
client {
  enabled = true
  
  reserved {
    cpu            = 4000
    memory         = 1000
  }
}
```
2) nomad agent -dev -config config.hcl
3) nomad plan example.nomad.hcl
```
+ Job: "example"
+ Task Group: "test" (1 create)
  + Task: "test" (forces create)

Scheduler dry-run:
- WARNING: Failed to place all allocations.
  Task Group "test" (failed to place 1 allocation):
    * Resources exhausted on 1 nodes
    * Dimension "cpu" exhausted on 1 nodes
```

![compute](https://github.com/hashicorp/nomad/assets/6938602/8633363f-916c-4690-bcc8-981848817445)

#### Expected Result

- Job has enought resourecs to be run

#### Actual Result

- Can't schedule job

### Job file (if appropriate)
```
job "example" {
  datacenters = ["*"]
  type = "service"
  group "test" {
    task "test" {
      driver = "raw_exec"
      config {
        command = ["/bin/true"]
      }
      resources {
        cpu    = 8000 
        memory = 256
      }
    }
  }
}
```

For this job to run on this node I have to set the cpu to 6400 for it to be allocated correctly (setting 6401 will not allow it). It seems like reserved CPU is taken into account two times.

**Comment by tgross** [comment]
Hi @jorgemarey! This does look troubling, thanks for all the detail you've provided here. I'm seeing what we can do to expedite a fix on this.

**Comment by pkazmierczak** [comment]
Hi @jorgemarey, thanks for reporting. I was able to reproduce this on Linux, but not on Darwin. Doesn't make it less of an issue, just noting here that so far this appears to be a Linux-specific bug. I'll continue investigating and keep you posted.

**Comment by pkazmierczak** [comment]
hey @jorgemarey, the patch landed in main and will be released in 1.8.2. Thanks again for reporting!
